spec/requests/api/v1/driver_commission_histories_spec.rb

```diff
@@ -1,24 +1,90 @@
 require 'rails_helper'
 
-RSpec.describe '/api/v1/all', :type => :request do
-  let(:all) do
-    DriverCommissionHistoryAdapter.all
-  end
+RSpec.describe '/api/v1/driver_commission_history', :type => :request do
+  describe 'all' do
+    let(:all) do
+      DriverCommissionHistoryAdapter.all
+    end
 
-  let(:by_paid_date) do
-    DriverCommissionHistoryAdapter.by_paid_date(Date.new(2012, 01, 01))
-  end
+    let(:all_url) { '/api/v1/driver_commissions_history' }
+
+    context 'when application id and secret are valid' do
+      let(:response) { doorkeeper_request.get(all_url) }
+      # it { expect(doorkeeper_request.token).to eq('')}
+      it 'responds with a body equal to json' do
+        expect(response.body).to eq(all.to_json)
+      end
+    end
 
-  it 'responds with all driver commission history json' do
-    get '/api/v1/driver_commissions_history'
+    context 'when application id and secret are not valid' do
```

Comment:
We may need to discuss this further as I am not sure now to create a get request with doorkeeper without the token instance. I would think the test is done as the token never gets generated if the uid and secret are wrong in the OAuth2::Client call.